clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp

```diff
@@ -424,51 +524,177 @@ class ObjectFileHandler final : public FileHandler {
   /// The name of the main input file.
   StringRef MainInputFileName;
 
-  /// Iterator of the current and next section.
-  section_iterator CurrentSection;
-  section_iterator NextSection;
-
 public:
   ObjectFileHandler(std::unique_ptr<ObjectFile> ObjIn)
       : FileHandler(), Obj(std::move(ObjIn)),
-        CurrentSection(Obj->section_begin()),
-        NextSection(Obj->section_begin()) {}
+        CurBundle(TripleToBundleInfo.end()),
+        NextBundle(TripleToBundleInfo.end()) {}
 
   ~ObjectFileHandler() final {}
 
-  void ReadHeader(MemoryBuffer &Input) final {}
+  // Iterate through sections and create a map from triple to relevant bundle
+  // information.
+  void ReadHeader(MemoryBuffer &Input) final {
+    for (section_iterator Sec = Obj->section_begin(); Sec != Obj->section_end();
```

Comment:
Perhaps an interface such as
```
for (auto const &Sec : Obj->sections()) {
```
or similar would be even clearer.

Reply:
I actually need section_iterator here, not SectionRef. So section_begin()/end() instead of sections()